src/Microsoft.AspNet.Authentication.OAuth/OAuthTokenResponse.cs

```diff
@@ -16,10 +16,26 @@ public OAuthTokenResponse(JObject response)
             ExpiresIn = response.Value<string>("expires_in");
         }
 
+        private OAuthTokenResponse(string error)
+        {
+            Error = error;
+        }
+
+        public static OAuthTokenResponse Success(JObject response)
+        {
+            return new OAuthTokenResponse(response);
+        }
+
+        public static OAuthTokenResponse Failed(string error)
+        {
+            return new OAuthTokenResponse(error);
+        }
+
         public JObject Response { get; set; }
         public string AccessToken { get; set; }
         public string TokenType { get; set; }
         public string RefreshToken { get; set; }
         public string ExpiresIn { get; set; }
+        public string Error { get; set; }
```

Comment:
Yeah, this looks weird. `error` is a standard OAuth2 parameter, but this property is just a user-defined exception.